frontend/online-shop/src/pages/products-collection/products-collection.page.tsx

```diff
@@ -0,0 +1,34 @@
+import { FC, useEffect, useState } from "react";
+import "./products-collection.style.css";
+import ProductPreview from "../../components/product-preview/product-preview.component";
+import { Grid } from "@mui/material";
+import { fetchProducts } from "../../services/products.service";
+const ProductsCollection: FC = () => {
+  // eslint-disable-next-line
+  const [product, setProduct] = useState([
```

Comment:
Why do you have a state called product at singular when you give it an initial value which is an array? This should be `products`. Also, don't give it an initial product that does not exist. Give it either `undefined` or `[]` and also show that loading indicator while the data loads